### PR TITLE
nspawn: change owner/group of /run/systemd/nspawn/notify to userns-root

### LINKED ISSUE #4944: systemd-nspawn -U --notify-ready=yes doesn't work
```sh
root# systemd-run --service-type=notify --unit=notify-test $SYSTEMD_ROOT/libtool --mode=execute $SYSTEMD_ROOT/systemd-nspawn --notify-ready=yes -D /var/tmp/systemd-test.fDEpDI/nspawn-root -U -b
Job for notify-test.service failed because a timeout was exceeded.
See "systemctl status notify-test3.service" and "journalctl -xe" for details.
```

```
# container's PID1
sendmsg(18<UNIX:[8888577]>, {msg_name={sa_family=AF_UNIX, sun_path="/run/systemd/nspawn/notify"}, msg_namelen=28, msg_iov=[{iov_base="READY=1\nSTATUS=Startup finished "..., iov_len=43}], msg_iovlen=1, msg
_controllen=0, msg_flags=0}, MSG_NOSIGNAL) = -1 EACCES (Permission denied)
```
```
# container
-bash-4.3# ls -l /run/systemd/nspawn/notify
srwxr-xr-x. 1 65534 65534 0 Dec 18 23:31 /run/systemd/nspawn/notify
```

https://github.com/systemd/systemd/pull/3474 contains
> This patch was also test with --private-users=yes, you can test it just adding it
at the end of the command at point 3

But I have no idea how did it work. @alepuccetti ?

### PR DESCRIPTION
**Pull request by evverx**
Closes #4944

### CODE REVIEW

File: src/nspawn/nspawn.c [hunk 1]
```diff
@@ -2363,6 +2363,10 @@ static int setup_sd_notify_child(void) {
                 return log_error_errno(errno, "bind(%s) failed: %m", sa.un.sun_path);
         }
 
+        r = userns_lchown(NSPAWN_NOTIFY_SOCKET_PATH, 0, 0);
+        if (r < 0)
+                return log_error_errno(r, "Failed to chown " NSPAWN_NOTIFY_SOCKET_PATH ": %m");
```
Comment (evverx) [inline src/nspawn/nspawn.c:2369]: I should `safe_close(fd)`

**Comment by evverx** [comment]
```
Jan 18 03:37:40 systemd-testsuite test-nspawn.sh[220]: nc: invalid option -- 'U'
Jan 18 03:37:40 systemd-testsuite test-nspawn.sh[220]: BusyBox v1.22.1 (Ubuntu 1:1.22.0-15ubuntu1) multi-call binary.
Jan 18 03:37:40 systemd-testsuite test-nspawn.sh[220]: Usage: nc [-iN] [-wN] [-l] [-p PORT] [-f FILE|IPADDR PORT] [-e PROG]
Jan 18 03:37:40 systemd-testsuite test-nspawn.sh[220]: Open a pipe to IP:PORT or FILE
Jan 18 03:37:40 systemd-testsuite test-nspawn.sh[220]:         -l        Listen mode, for inbound connects
Jan 18 03:37:40 systemd-testsuite test-nspawn.sh[220]:                 (use -ll with -e for persistent server)
Jan 18 03:37:40 systemd-testsuite test-nspawn.sh[220]:         -p PORT        Local port
Jan 18 03:37:40 systemd-testsuite test-nspawn.sh[220]:         -w SEC        Connect timeout
Jan 18 03:37:40 systemd-testsuite test-nspawn.sh[220]:         -i SEC        Delay interval for lines sent
Jan 18 03:37:40 systemd-testsuite test-nspawn.sh[220]:         -f FILE        Use file (ala /dev/ttyS0) instead of network
Jan 18 03:37:40 systemd-testsuite test-nspawn.sh[220]:         -e PROG        Run PROG after connect
Jan 18 03:37:40 systemd-testsuite test-nspawn.sh[220]: Container nc-container failed with error code 1.
```
I should change `nc -U --udp` to `full-path-to-nc -U --udp`
@martinpitt , just to be sure. Which version of `nc` are we using on testbeds?

**Comment by martinpitt** [comment]
.. oh, and while you are at it, fix the "onwer" typo in the commit message :-)

**Comment by evverx** [comment]
@martinpitt , done. Thanks!

**Pull request merged**